mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/index/MongoDBIndexSpec.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.index;
```

Comment:
How comes this is in the API? I don't think we expect the user to interact with that. If so, it should go somewhere into _impl_.